spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/core/CredentialsProviderAutoConfiguration.java

```diff
@@ -73,6 +83,11 @@ public static AwsCredentialsProvider createCredentialsProvider(CredentialsProper
 			providers.add(createProfileCredentialProvider(profile));
 		}
 
+		StsProperties sts = properties.getSts();
+		if (ClassUtils.isPresent(STS_WEB_IDENTITY_TOKEN_FILE_CREDENTIALS_PROVIDER, null) && sts != null) {
```

Comment:
Perhaps we should not check for `sts != null` and return `StsWebIdentityTokenFileCredentialsProvider` if sts is on the classpath.